File: backend/src/modules/transportation/repricing/repricing.controller.ts
import { Body, Controller, Get, Headers, Post, UseGuards } from '@nestjs/common';
import { HeaderAuthGuard, Validation } from '../../../common/auth/header-auth.guard';
import { CurrentUser } from '../../../common/auth/current-user.decorator';
import { RepricingService } from './repricing.service';

// Base path /api/Transportation (faithful to the documented URLs); every action
// requires the CompanyName + UserToken headers via the guard. Filters/pagination
// ride in HTTP headers (Node lowercases them), bodies are JSON, and
// approvals/rejections are POSTs with an Id header — exactly as documented (§6.3).
@Controller('api/Transportation')
@UseGuards(HeaderAuthGuard)
export class RepricingController {
  constructor(private readonly service: RepricingService) {}

  @Post('ModifyPriceOfTransportationLine')
  modify(@Body() body: RepriceBody, @CurrentUser() v: Validation) {
    return this.service.modify(body, v.userId);
  }

  @Get('getAllModifyPriceOfTransportationLine')
  getAll(
    @Headers('pageno') pageNo?: string,
    @Headers('noofitems') noOfItems?: string,
    @Headers('approve') approve?: string,
  ) {
    const approveFilter =
      approve === undefined || approve === '' ? undefined : String(approve).toLowerCase() === 'true';
    return this.service.getAll(Number(pageNo) || 1, Number(noOfItems) || 20, approveFilter);
  }

  @Post('UpdatePriceOfTransportationLine')
  approvePrice(@Headers('id') id: string, @CurrentUser() v: Validation) {
    return this.service.approvePrice(Number(id), v.userId);
  }

  @Post('RejectUpdatePrice')
  rejectPrice(@Headers('id') id: string, @CurrentUser() v: Validation) {
    return this.service.rejectPrice(Number(id), v.userId);
  }
}

// Shape of the ModifyPriceOfTransportationLine payload (documented PascalCase keys, §6.3).
export interface RepriceBody {
  IncreaseCost?: number;
  IsPercent?: boolean;
  ForAllLines?: boolean;
  ApproximateToFiveFlag?: boolean;
  TransportationLineIds?: number[]; // ARRAY of ROUTE ids
  StartDate?: string; // ISO-UTC
}

File: backend-dotnet/GarasTransport.Api/Modules/Transportation/Repricing/RepricingController.cs
using GarasTransport.Api.Common;
using Microsoft.AspNetCore.Mvc;

namespace GarasTransport.Api.Modules.Transportation.Repricing;

// Base path /api/Transportation (§6.3). Every action requires CompanyName +
// UserToken headers via the filter.
[Route("api/Transportation")]
[HeaderAuth]
public class RepricingController : ApiControllerBase
{
    private readonly RepricingService _service;

    public RepricingController(RepricingService service) => _service = service;

    [HttpPost("ModifyPriceOfTransportationLine")]
    public Task<object> Modify([FromBody] RepriceBody body) => _service.Modify(body ?? new(), CurrentUser.UserId);

    [HttpGet("getAllModifyPriceOfTransportationLine")]
    public Task<object> GetAll()
    {
        var raw = Header("Approve");
        bool? approveFilter = string.IsNullOrEmpty(raw) ? null : string.Equals(raw, "true", StringComparison.OrdinalIgnoreCase);
        return _service.GetAll(HeaderInt("PageNo", 1), HeaderInt("NoOfItems", 20), approveFilter);
    }

    [HttpPost("UpdatePriceOfTransportationLine")]
    public Task<object> ApprovePrice() => _service.ApprovePrice(HeaderInt("Id", 0), CurrentUser.UserId);

    [HttpPost("RejectUpdatePrice")]
    public Task<object> RejectPrice() => _service.RejectPrice(HeaderInt("Id", 0), CurrentUser.UserId);
}
